scout/server/blueprints/clinvar/controllers.py

```diff
@@ -325,12 +327,79 @@ def update_clinvar_submission_status(request_obj, institute_id, submission_id):
         )
 
 
+def validate_submission(submission_id):
+    """Validate a submission object and documents using the ClinVar API
+
+    Args:
+        submission_id(str): the database id of a clinvar submission
+    """
+
+    variant_data = store.clinvar_objs(submission_id, "variant_data")
+    obs_data = store.clinvar_objs(submission_id, "case_data")
+    if not variant_data or not obs_data:
+        flash("Submission must contain both Variant and CaseData info", "warning")
+        return
+
+    # Retrieve eventual assertion criteria for the submission
+    extra_params = store.clinvar_assertion_criteria(variant_data[0]) or {}
+
+    # Retrieve genome build for the case submitted
+    case_obj = store.case(case_id=variant_data[0].get("case_id")) or {"genome_build": 37}
+    extra_params["assembly"] = "GRCh37" if "37" in str(case_obj.get("genome_build")) else "GRCh38"
+
+    def _write_file(afile, header, lines):  # Write temp CSV file
+        writes = csv.writer(afile, delimiter=",", quoting=csv.QUOTE_ALL)
+        writes.writerow(header)
+        for line in lines:
+            writes.writerow(line)
+        afile.flush()
+        afile.seek(0)
+
+    with NamedTemporaryFile(
+        mode="a+", prefix="Variant", suffix=".csv"
+    ) as variant_file, NamedTemporaryFile(
+        mode="a+", prefix="CaseData", suffix=".csv"
+    ) as casedata_file:
+
+        # Write temp Variant CSV file
+        _, variants_header, variants_lines = clinvar_submission_file(
+            submission_id, "variant_data", "SUB000"
+        )
+        _write_file(variant_file, variants_header, variants_lines)
+
+        # Write temp CaseData CSV file
+        _, casedata_header, casedata_lines = clinvar_submission_file(
+            submission_id, "case_data", "SUB000"
+        )
+        _write_file(casedata_file, casedata_header, casedata_lines)
+
+        code, conversion_res = clinvar_api.convert_to_json(
+            variant_file.name, casedata_file.name, extra_params
+        )
+
+        if code != 200:  # Connection or conversion object errors
+            flash(str(conversion_res), "warning")
+            return
+
+        code, valid_res = clinvar_api.validate_json(
+            subm_data=conversion_res.json(), api_key=current_app.config.get("CLINVAR_API_KEY")
+        )
+
+        if code != 201:  # Connection or conversion object errors
+            flash(str(valid_res.__dict__), "warning")
```

Comment:
I wrote it like this so it catches errors of different type, for instance connection errors or wrong API key errors. I am afraid if you start to parse messages of a certain type too well it will crash on other occasions

Reply:
Think about errors not in the ClinVar API, but in preClinVar. I found and fixed a bunch of them in the last week

Reply:
I guess it is also a little different now that we wrap the editing part for the users. If they can't edit these csv files, it makes as much sense just to have the error to post as an issue... 😆